src/PostgREST/App.hs

```diff
@@ -376,18 +377,20 @@ app dbStructure proc cols conf apiRequest =
           (readRequestToQuery <$> readReq) <*>
           (mutateRequestToQuery <$> mutReq)
 
-responseContentTypeOrError :: [ContentType] -> [ContentType] -> Action -> Target -> Either Response ContentType
-responseContentTypeOrError accepts rawContentTypes action target = serves contentTypesForRequest accepts
+responseContentTypeOrError :: Maybe Text -> [ContentType] -> [ContentType] -> Action -> Target -> Either Response ContentType
+responseContentTypeOrError procAccept accepts rawContentTypes action target = serves contentTypesForRequest accepts
   where
     contentTypesForRequest = case action of
       ActionRead _       ->  [CTApplicationJSON, CTSingularJSON, CTTextCSV]
                              ++ rawContentTypes
       ActionCreate       ->  [CTApplicationJSON, CTSingularJSON, CTTextCSV]
       ActionUpdate       ->  [CTApplicationJSON, CTSingularJSON, CTTextCSV]
       ActionDelete       ->  [CTApplicationJSON, CTSingularJSON, CTTextCSV]
-      ActionInvoke _     ->  [CTApplicationJSON, CTSingularJSON, CTTextCSV]
-                             ++ rawContentTypes
-                             ++ [CTOpenAPI | tpIsRootSpec target]
+      ActionInvoke _     ->  case procAccept of
+                               Just acc -> [CTOther $ toS acc]
```

Comment:
This looks like it could easily be extended to allow multiple media types, e.g. like `pgrst.accept='image/png,image/jpeg'` - would just need to convert this to a list of multiple `CTOther`?